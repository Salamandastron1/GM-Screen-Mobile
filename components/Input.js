import React from 'react';
import { TextInput, View } from 'react-native';
import PropTypes from 'prop-types';

const Input = ({ value, onChange, name }) => (
  <View style={styles.containerBox}>
    <TextInput
      value={value}
      onChange={onChange}
      style={styles.inputBox}
      name={name}
    />
  </View>
);

const styles = {
  inputBox: {
    height: 40,
  },
  containerBox: {
    flex: 1,
    backgroundColor: '#54b8c0',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#54b8c0',
    alignItems: 'stretch',
    marginLeft: 10,
    marginRight: 10,
  },
};

Input.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};

export default Input;
