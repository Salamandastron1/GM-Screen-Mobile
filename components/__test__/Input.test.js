/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input';

it('should match the shapshot', () => {
  const tree = shallow(<Input />);
  expect(tree).toMatchSnapshot();
});