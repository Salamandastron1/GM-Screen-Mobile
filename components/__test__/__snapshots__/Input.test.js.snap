// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the shapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#54b8c0",
      "borderColor": "#54b8c0",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "marginLeft": 10,
      "marginRight": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "height": 40,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
  />
</Component>
`;
